lib/vmpooler/pool_manager.rb

```diff
@@ -267,10 +267,34 @@ def clone_vm(pool_name, provider)
       end
     end
 
-    def _clone_vm(pool_name, provider)
+    def generate_and_check_hostname(pool_name)
       # Generate a randomized hostname
       random_name = [@name_generator.adjective(max: 7), @name_generator.noun(max: 7)].join('-')
-      new_vmname = $config[:config]['prefix'] + random_name
+      hostname = $config[:config]['prefix'] + random_name
+      available = $redis.hlen('vmpooler__vm__' + hostname) == 0
```

Comment:
To test this out, I hardcoded `testing-testing` as the `random_name` and observed `_clone_vm` repeatedly failing to provision. 
```
vmpooler_1    | [2019-11-04 19:06:01] [*] [snapshot_manager] starting worker thread[2019-11-04 19:06:01] [vsphere-vc6] Connection Pool - Creating a connection object
vmpooler_1    |
vmpooler_1    | [2019-11-04 19:06:01] [*] [centos-7-x86_64] starting worker thread
vmpooler_1    | [2019-11-04 19:06:03] [vsphere-vc6] Connection Pool - Creating a connection object
vmpooler_1    | [2019-11-04 19:06:03] [?] [centos-7-x86_64] 'testing-testing' added to 'discovered' queue
vmpooler_1    | [2019-11-04 19:06:03] [!] [centos-7-x86_64] is empty
vmpooler_1    | [2019-11-04 19:06:03] [ ] [centos-7-x86_64] Starting to clone 'testing-testing'
vmpooler_1    | [2019-11-04 19:06:03] [!] [centos-7-x86_64] Generated hostname testing-testing was not unique (attempt #1 of 3)
vmpooler_1    | [2019-11-04 19:06:03] [!] [centos-7-x86_64] Generated hostname testing-testing was not unique (attempt #2 of 3)
vmpooler_1    | [2019-11-04 19:06:03] [!] [centos-7-x86_64] failed while cloning VM with an error: Unable to generate a unique hostname after 3 attempts. The last hostname checked was testing-testing
vmpooler_1    | warning: thread "Ruby-0-Thread-12: /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:260" terminated with exception (report_on_exception is true):
vmpooler_1    | RuntimeError: Unable to generate a unique hostname after 3 attempts. The last hostname checked was testing-testing
vmpooler_1    |   find_unique_hostname at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:292
vmpooler_1    |              _clone_vm at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:298
vmpooler_1    |               clone_vm at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:262
vmpooler_1    | [2019-11-04 19:06:04] [!] [centos-7-x86_64] failed while cloning VM with an error: DuplicateName: The name 'testing-testing' already exists.
vmpooler_1    | warning: thread "Ruby-0-Thread-11: /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:260" terminated with exception (report_on_exception is true):
vmpooler_1    | RbVmomi::Fault: DuplicateName: The name 'testing-testing' already exists.
vmpooler_1    |   wait_for_completion at /usr/local/bundle/gems/rbvmomi-1.13.0/lib/rbvmomi/vim/Task.rb:14
vmpooler_1    |             create_vm at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/providers/vsphere.rb:352
vmpooler_1    |          with_metrics at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/generic_connection_pool.rb:27
vmpooler_1    |      handle_interrupt at org/jruby/RubyThread.java:760
vmpooler_1    |          with_metrics at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/generic_connection_pool.rb:26
vmpooler_1    |      handle_interrupt at org/jruby/RubyThread.java:760
vmpooler_1    |          with_metrics at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/generic_connection_pool.rb:19
vmpooler_1    |             create_vm at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/providers/vsphere.rb:284
vmpooler_1    |             _clone_vm at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:308
vmpooler_1    |              clone_vm at /usr/local/bundle/gems/vmpooler-0.8.1/lib/vmpooler/pool_manager.rb:262
vmpooler_1    | [2019-11-04 19:06:09] [!] [centos-7-x86_64] Generated hostname testing-testing was not unique (attempt #1 of 3)
vmpooler_1    | [2019-11-04 19:06:09] [!] [centos-7-x86_64] Generated hostname testing-testing was not unique (attempt #1 of 3)
vmpooler_1    | [2019-11-04 19:06:09] [!] [centos-7-x86_64] Generated hostname testing-testing was not unique (attempt #2 of 3)
vmpooler_1    | [2019-11-04 19:06:09] [!] [centos-7-x86_64] Generated hostname testing-testing was not unique (attempt #2 of 3)
```